apple/internal/framework_import_support.bzl

```diff
@@ -155,28 +151,73 @@ def _classify_framework_imports(config_vars, framework_imports):
             continue
 
         # Path matching
-        if "Headers" in file.short_path:
+        if "Headers/" in file.short_path:
             header_imports.append(file)
             continue
 
-        # Unknown file type, sending tu bundling (i.e. resources)
+        # Unknown file type, sending to unknown (i.e. resources, Info.plist, etc.)
+        unknown_imports.append(file)
+
+    return struct(
+        header_imports = header_imports,
+        module_map_imports = module_map_imports,
+        swift_module_imports = swift_module_imports,
+        unknown_imports = unknown_imports,
+    )
+
+def _classify_framework_imports(config_vars, framework_imports):
+    """Classify a list of files referencing an Apple framework.
+
+    Args:
+        config_vars: A dictionary (String to String) of configuration variables. Can be from ctx.var.
+        framework_imports: List of File for an imported Apple framework.
+    Returns:
+        A struct containing classified framework import files by categories:
+            - bundle_name: The framework bundle name infered by filepaths.
+            - binary_imports: Apple framework binary imports.
+            - bundling_imports: Apple framework bundle imports.
+            - header_imports: Apple framework header imports.
+            - module_map_imports: Apple framework modulemap imports.
+            - swift_module_imports: Apple framework swiftmodule imports.
+    """
+    framework_imports_by_category = _classify_file_imports(config_vars, framework_imports)
+
+    bundle_name = None
+    bundling_imports = []
+    binary_imports = []
+    for file in framework_imports_by_category.unknown_imports:
+        # Infer framework bundle name and binary
+        parent_dir_name = paths.basename(file.dirname)
+        is_bundle_root_file = parent_dir_name.endswith(".framework")
+        if is_bundle_root_file:
+            bundle_name, _ = paths.split_extension(parent_dir_name)
+            if file.basename == bundle_name:
+                binary_imports.append(file)
+                continue
+
         bundling_imports.append(file)
 
+    # TODO: Enable these checks once static library support works with them
```

Comment:
Is this Google's todo or ours?

Reply:
i added this one, right now since this is shared with our old static impl this check fails, but i will fix that as more cherry picks land